tests/integration/test_kinesis.py

```diff
@@ -305,6 +308,104 @@ def test_record_lifecycle_data_integrity(
         for response_record in response_records:
             assert response_record.get("Data").decode("utf-8") in records_data
 
+    @pytest.mark.aws_validated
+    @patch.object(kinesis_listener, "MAX_SUBSCRIPTION_SECONDS", 3)
+    def test_subscribe_to_shard_timeout(
+        self,
+        kinesis_client,
+        kinesis_create_stream,
+        wait_for_stream_ready,
+        wait_for_consumer_ready,
+        snapshot,
+    ):
+        # create stream and consumer
+        stream_name = kinesis_create_stream(ShardCount=1)
+        stream_arn = kinesis_client.describe_stream(StreamName=stream_name)["StreamDescription"][
+            "StreamARN"
+        ]
+        wait_for_stream_ready(stream_name)
+        # create consumer
+        result = kinesis_client.register_stream_consumer(StreamARN=stream_arn, ConsumerName="c1")[
+            "Consumer"
+        ]
+        consumer_arn = result["ConsumerARN"]
+        wait_for_consumer_ready(consumer_arn=consumer_arn)
+
+        # subscribe to shard
+        response = kinesis_client.describe_stream(StreamName=stream_name)
+        shard_id = response.get("StreamDescription").get("Shards")[0].get("ShardId")
+        result = kinesis_client.subscribe_to_shard(
+            ConsumerARN=result["ConsumerARN"],
+            ShardId=shard_id,
+            StartingPosition={"Type": "TRIM_HORIZON"},
+        )
+        stream = result["EventStream"]
+
+        # letting the subscription run out
+        time.sleep(5)
+
+        # put records
+        msg = b"Hello world"
+        kinesis_client.put_records(
+            StreamName=stream_name, Records=[{"Data": msg, "PartitionKey": "1"}]
+        )
+
+        # due to the subscription being timed out, we should not be able to read out results
+        results = []
+        for entry in stream:
+            records = entry["SubscribeToShardEvent"]["Records"]
+            results.extend(records)
+
+        snapshot.match("Records", results)
+
+        # clean up
+        kinesis_client.deregister_stream_consumer(StreamARN=stream_arn, ConsumerName="c1")
+
+    def test_subscribe_to_shard_pings(self):
+        pass
```

Comment:
This one seems to be a leftover, or a stub for a test not yet implemented?

Reply:
removed it :)